intake_bluesky/core.py

```diff
@@ -889,3 +888,117 @@ def parse_handler_registry(handler_registry):
+                           for key in data_keys},
+            'filled': {key: xarray.concat([page['filled'][key] for page in pages])
+                       for key in data_keys}}
+
+
+def event_page_to_dataarray_page(event_page, dims=None, coords=None):
+    """
+    Converts the event_page's data, timestamps, and filled to xarray.DataArray.
+
+    Parameters
+    ----------
+    event_page: dict
+        A EventPage document
+    dims: tuple
+        Tuple of dimension names associated with the array
+    coords: dict-like
+        Dictionary-like container of coordinate arrays
+    Returns
+    ------
+    event_page : dict
+        An event_pages with xarray.dataArrays in the data,
+        timestamp, and filled fields.
+    """
+    if coords is None:
+        coords = {'time': event_page['time']}
+    if dims is None:
+        dims = ('time',)
+
+    array_keys = ['seq_num', 'time', 'uid']
+    data_keys = event_page['data'].keys()
+
+    return {'descriptor': event_page['descriptor'],
+            **{key: event_page[key] for key in array_keys},
+            'data': {key: xarray.DataArray(
+                            event_page['data'][key], dims=dims, coords=coords, name=key)
+                     for key in data_keys},
+            'timestamps': {key: xarray.DataArray(
+                            event_page['timestamps'][key], dims=dims, coords=coords, name=key)
+                           for key in data_keys},
+            'filled': {key: xarray.DataArray(
+                            event_page['filled'][key], dims=dims, coords=coords, name=key)
+                       for key in data_keys}}
+
+
+def dataarray_page_to_dataset_page(dataarray_page):
+
+    """
+    Converts the dataarray_page's data, timestamps, and filled to xarray.DataSet.
+
+    Parameters
+    ----------
+    dataarray_page: dict
+    Returns
+    ------
+    dataset_page : dict
+    """
+    array_keys = ['seq_num', 'time', 'uid']
+
+    return {'descriptor': dataarray_page['descriptor'],
+            **{key: dataarray_page[key] for key in array_keys},
+            'data': xarray.merge(dataarray_page['data'].values()),
+            'timestamps': xarray.merge(dataarray_page['timestamps'].values()),
+            'filled': xarray.merge(dataarray_page['filled'].values())}
+
+
+class DaskFiller(event_model.Filler):
+
+    def __init__(self, *args, inplace=False, **kwargs):
+        super().__init__(*args, inplace=False, **kwargs)
+
+    def event_page(self, doc):
+
+        @dask.delayed
+        def delayed_fill(event_page, key):
+            filled_page = self.fill_event_page(event_page, include=key)
+            return numpy.asarray(filled_page['data'][key])
+
+        descriptor = self._descriptor_cache[doc['descriptor']]
+        needs_filling = {key for key, val in descriptor['data_keys'].items()
+                         if 'external' in val}
+
+        for key in needs_filling:
+            doc['data'][key] = array.from_delayed(delayed_fill(doc, key))
```

Comment:
This is modifying `doc` in place. We need to take the same approach you took in `event_model.Filler`.